testng-core/src/test/kotlin/test/attributes/samples/custom/CustomAttributesTransformer.kt

```diff
@@ -0,0 +1,36 @@
+package test.attributes.samples.custom
+
+import java.lang.reflect.Constructor
+import java.lang.reflect.Method
+import org.testng.IAnnotationTransformer
+import org.testng.annotations.CustomAttribute
+import org.testng.annotations.ITestAnnotation
+import kotlin.reflect.KCallable
+
+class CustomAttributesTransformer : IAnnotationTransformer {
+
+    @Override
+    override fun transform(
+        annotation: ITestAnnotation?,
+        testClass: Class<*>?,
+        testConstructor: Constructor<*>?,
+        testMethod: Method?
+    ) {
+        val attributes = arrayOf(
+            customAttribute("sorrow", arrayOf("Coffee", "Tea")))
+        annotation?.attributes = attributes
+    }
+
+    // Support for creation annotation literals is still not there in Kotlin.
+    // Refer to the Open Defect https://youtrack.jetbrains.com/issue/KT-25947
+    // Below work-around is the only way in which we can create annotation literals in Kotlin
+    // and has been borrowed from https://stackoverflow.com/q/64539969
+    private fun customAttribute(name:String, values: Array<String>) : CustomAttribute{
+        val constructors = CustomAttribute::class.constructors
+        val callable: KCallable<*> = constructors.first()
+        val param1 = callable.parameters[0]
+        val param2 = callable.parameters[1]
+        return callable.callBy(mapOf(param1 to name, param2 to values)) as CustomAttribute
+    }
```

Comment:
This is not needed with Kotlin 1.6+: https://kotlinlang.org/docs/whatsnew1620.html#instantiation-of-annotation-classes

Reply:
Bit more idiomatic Kotlin (avoid repeating `callable` multiple times:

    with (CustomAttribute::class.constructors.first()) {
        // "this" is now what used to be callable
        val param1 = parameters[0]
        val param2 = parameters[1]
        return callBy(mapOf(param1 to name, param2 to values)) as CustomAttribute
    }